port/boards/rv32emu/main.c

```diff
@@ -52,14 +81,59 @@ void *qembd_allocmain(size_t size)
 	return malloc(size);
 }
 
+static int poll_event()
+{
+    register int a0 asm("a0") = (uintptr_t) &event;
+    register int a7 asm("a7") = 0xc0de;
+    asm volatile("scall" : "+r"(a0) : "r"(a7));
+    if (event.type == MOUSE_MOTION_EVENT) {
+        mouse_motion.xrel += event.mouse.motion.xrel;
+        mouse_motion.yrel += event.mouse.motion.yrel;
+    }
+    return a0;
+}
+
 int qembd_dequeue_key_event(key_event_t *e)
 {
-	/* Not Implemented */
-	return -1;
+	while (poll_event()) {
+		if (event.type == KEY_EVENT) {
+			*e = event.key_event;
+			/* remap keyode */
+			switch (e->keycode) {
```

Comment:
What if caps lock is on?

Reply:
Caps lock has no effect on these keycodes because keys with printable characters are represented by the corresponding lowercase letter, as listed in [SDL Keycode Lookup Table](https://wiki.libsdl.org/SDLKeycodeLookup), there are no uppercase keycodes because keys are just keys.